src/x3f_image.c

```diff
@@ -115,9 +144,17 @@
 {
   uint32_t keep[4], keep_cols, keep_rows;
 
-  if (!x3f_get_camf_matrix(x3f, name, 4, 0, 0, M_UINT, rect)) return 0;
+  if (!strncmp(name, "Fake", 4)) {
+    if (!x3f_get_fake_camf_matrix(x3f, name, image, rect))
+      return 0;
+  } else {
+    if (!x3f_get_camf_matrix(x3f, name, 4, 0, 0, M_UINT, rect))
+      return 0;
+  }
+
```

Comment:
A thought ... but even that would be a kludge.

I first started to do something in the get_black_level function, where I would call different get_rect functions and then use the rects as arguments. Then there would be nu ugliness in the library functions. But, then I could not reuse your thinking regarding the keep_areas and mappings from rects to crops. I then had to reinvent all that stuff.